core/src/main/java/org/elasticsearch/index/mapper/TypeFieldMapper.java

```diff
@@ -88,29 +92,12 @@ public MetadataFieldMapper getDefault(MappedFieldType fieldType, ParserContext c
     }
 
     static final class TypeFieldType extends StringFieldType {
-        private boolean fielddata;
```

Comment:
Also we won't be able to open indices that had types with fielddata in 6.0 then. All fine with me so long as we deprecate it in 5.x.